projects/packages/search/src/dashboard/components/dashboard/index.jsx

```diff
@@ -0,0 +1,186 @@
+/**
+ * External dependencies
+ */
+import React, { Fragment, useMemo } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch, select as syncSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import analytics from '@automattic/jetpack-analytics';
+import restApi from '@automattic/jetpack-api';
+import { JetpackFooter, JetpackLogo, Spinner } from '@automattic/jetpack-components';
+import ModuleControl from 'components/module-control';
+import MockedSearch from 'components/mocked-search';
+import { STORE_ID } from 'store';
+import NoticesList from 'components/global-notices';
+
+import 'scss/rna-styles.scss';
+import './style.scss';
+
+/**
+ * SearchDashboard component definition.
+ *
+ * @returns {React.Component} Search dashboard component.
+ */
+export default function SearchDashboard() {
+	useSelect( select => select( STORE_ID ).getSearchPlanInfo(), [] );
+	useSelect( select => select( STORE_ID ).getSearchModuleStatus(), [] );
+
+	const siteAdminUrl = syncSelect( STORE_ID ).getSiteAdminUrl();
+	const aboutPageUrl = siteAdminUrl + 'admin.php?page=jetpack_about';
+
+	const updateOptions = useDispatch( STORE_ID ).updateJetpackSettings;
+	const isInstantSearchPromotionActive = syncSelect( STORE_ID ).isInstantSearchPromotionActive();
+
+	const domain = syncSelect( STORE_ID ).getCalypsoSlug();
+	const upgradeBillPeriod = syncSelect( STORE_ID ).getUpgradeBillPeriod();
```

Comment:
My intention to use syncSelect on static data is that it doesn't subscribe state changes, which could potentially save a little bit of resource.

Reply:
Due to the nature of functional components, I believe this will actually have the opposite effect of what you intend. Invocations of `syncSelect` will be executed on every render; however, invocations of [`useSelect`](https://github.com/WordPress/gutenberg/tree/trunk/packages/data#useselect) with the correct `dependencies` parameter will only execute when the dependencies have updated. 

See [this section on performance optimization](https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects) when using hooks.

Reply:
Hmm as I understand how functional components work - when the subscribed states change, React would trigger a re-render of the component, which is actually a call to the function. Rather than just select the value from the store, with `useSelect` it  just hooks on state changes. That said, on every re-rendering, states are all selected again from the store. 

`useSelect` with `dependencies` will only trigger more re-rendering, because the state will not only triggering changes when the state changes, but also on the dependencies changes.

Am I getting anything wrong here? Thanks

Reply:
However the effect to use `syncSelect` is very minimum, which rather causes more confusion than the very limit gains.

Reply:
I think I understand what you're saying. To summarize:

- `useSelect` will execute its first parameter on every render as well as every state change.
- `syncSelect` will execute on every render, but ignore state changes.

Therefore, we're using `syncSelect` for these values since they're not dependent on state changes. Is this correct? If so, let's at least add in a comment explaining this to avoid any future confusion.

Reply:
That's right, thanks for the conclusion. I'll just change them to `useSelect`.
Addressed in e873e3ed2f